### ISSUE TITLE
[4.x]Make pack_fopen a weak symbol, add __allegro_pack_fopen

### ISSUE THREAD

**Issue by allebot** [liballeg/allegro5#460]
Originally reported by: bero

Original Ticket: [alleg/feature-requests/42](https://sourceforge.net/p/alleg/feature-requests/42)

When porting AGS to Linux, I noticed they copied src/file.c from allegro in order to rename pack_fopen to __old_pack_fopen and override it with a new pack_fopen that handles looking inside another file (and then falls back to calling __old_pack_fopen).

Since this is a relatively dirty hack and doesn't work with static linking at all (and I can see valid reasons to do it), I'd propose to rename pack_fopen to __allegro_pack_fopen and adding a pack_fopen that just calls __allegro_pack_fopen, but is declared a weak symbol.

This way:
- "Normal" applications using allegro will keep working just fine
- AGS and others wanting to do the same thing can override pack_fopen without having to duplicate code
- Overridden pack_fopen functions can call the original (because it's available unconditionally through the __allegro_pack_fopen symbol)

**Comment by ivan-mogilko** [comment]
I was recently doing investigation on this in the project that caused this feature request (Adventure Game Studio), and the actual problem, at least according to what I found so far, is that SOME of the Allegro objects do not (or did not) support creation from passed PACKFILE, only from passed filename.

Primary issue for our project is "load_midi". I checked in 4.4 branch and there is still no variant that would create MIDI object from passed PACKFILE. I believe it should not be a problem to add such function to Allegro API, because load_midi calls pack_fopen internally anyway.
